Source/CarthageKit/Project.swift

```diff
@@ -292,6 +318,107 @@ public final class Project {
 			.then(checkoutResolvedDependencies())
 	}
 
+	/// Installs binaries for the given project, if available.
+	///
+	/// Sends a boolean indicating whether binaries were installed.
+	private func installBinariesForProject(project: ProjectIdentifier, atRevision revision: String) -> ColdSignal<Bool> {
+		return ColdSignal.lazy {
+			if !self.useBinaries {
+				return .single(false)
+			}
+
+			let checkoutDirectoryURL = self.directoryURL.URLByAppendingPathComponent(project.relativePath, isDirectory: true)
+
+			switch project {
+			case let .GitHub(repository):
+				return self.downloadMatchingBinariesForProject(project, atRevision: revision, fromRepository: repository, withCredentials: nil)
+					.catch { error in
+						// If we were unable to fetch releases, try loading credentials from Git.
+						if error.domain == NSURLErrorDomain {
+							return GitHubCredentials.loadFromGit()
+								.mergeMap { credentials in
+									if let credentials = credentials {
+										return self.downloadMatchingBinariesForProject(project, atRevision: revision, fromRepository: repository, withCredentials: credentials)
+									} else {
+										return .error(error)
+									}
+								}
+						} else {
+							return .error(error)
+						}
+					}
+					.concatMap(unzipArchiveToTemporaryDirectory)
+					.concatMap { directoryURL in
+						return frameworksInDirectory(directoryURL)
+							.mergeMap(self.copyFrameworkToBuildFolder)
+							.on(completed: {
+								NSFileManager.defaultManager().trashItemAtURL(checkoutDirectoryURL, resultingItemURL: nil, error: nil)
+								return
```

Comment:
I learned the other day that you can do:

```swift
_ = expression…
```

in order to conveniently return `()` instead of the type of `expression`.